### PR TITLE
Cleanup, dev docs, code docs

### PR DESCRIPTION
**Pull request by m-kuhn**
A bit of housekeeping :broom: 

- [x] check proj data dir on android
- [x] check proj data dir on ios
- [x] check app starts properly
- [x] remove world map
- [x] rename systemgenericdatalocation to systemshareddayalocation and implement a default relative to the bin patb, override on android.

Fixes https://github.com/opengisch/QField/issues/2147

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d89129e2148c2766c19dab97e1fc820a30807207: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3117-3933a6e71084f6a7ac59ceaa5c81371df6b495e7-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3117-3933a6e71084f6a7ac59ceaa5c81371df6b495e7-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3117-3933a6e71084f6a7ac59ceaa5c81371df6b495e7-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3117-3933a6e71084f6a7ac59ceaa5c81371df6b495e7-x86-android.apk)
</details>

**Comment by m-kuhn** [comment]
@nirvn can you test if this branch builds fine on your system in a clean build folder? Just to be sure system dependencies are still found with the CMake changes.

**Comment by nirvn** [comment]
@m-kuhn , fails on Ubuntu 22.04:

```
CMake Error at CMakeLists.txt:128 (find_package):
  By not providing "FindPROJ.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "PROJ", but
  CMake did not find one.

  Could not find a package configuration file provided by "PROJ" with any of
  the following names:

    PROJConfig.cmake
    proj-config.cmake

  Add the installation prefix of "PROJ" to CMAKE_PREFIX_PATH or set
  "PROJ_DIR" to a directory containing one of the above files.  If "PROJ"
  provides a separate development package or SDK, be sure it has been
  installed.
```

I think you've tried to remove this findproj.cmake file once and it failed then. Still fails now :(

**Comment by m-kuhn** [comment]
Ah right, now I remember again, proj is built using autotools instead of cmake on ubuntu, that's why.

**Comment by m-kuhn** [comment]
> I think you've tried to remove this findproj.cmake file once and it failed then. Still fails now :(

Should work now, it's a more modern version (taken from GDAL) which produces a cmake target (no more specifying the include dir), also has a package name and target capitalization `PROJ` that's compatible with the config that's produced by proj itself, so we can remove that when time has come.

File: src/app/main.cpp [hunk 1]
```diff
@@ -166,56 +174,35 @@ int main( int argc, char **argv )
...
 #ifdef RELATIVE_PREFIX_PATH
+  app.setPrefixPath( app.applicationDirPath() + "/..", true );
   qputenv( "GDAL_DATA", QDir::toNativeSeparators( app.applicationDirPath() + "/../share/gdal" ).toLocal8Bit() );
   const QString projPath( QDir::toNativeSeparators( app.applicationDirPath() + "/../share/proj" ) );
-  app.setPrefixPath( app.applicationDirPath() + "/..", true );
 #else
-  const QString projPath;
-  app.setPrefixPath( CMAKE_INSTALL_PREFIX, true );
-#endif
+  const QString projPath = PlatformUtilities::instance()->systemGenericDataLocation() + QStringLiteral( "/proj/data" );
```
Comment (nirvn) [inline src/app/main.cpp:203]: @m-kuhn , that's not working on linux when building against distro installed proj.

**Comment by nirvn** [comment]
@m-kuhn , I can build now, but there's an issue with proj.db not found (commented above), as well as QGIS plugins not registering, leading to failure to load WMS providet et cie:
image

File: src/app/main.cpp [hunk 2]
```diff
@@ -166,56 +174,35 @@ int main( int argc, char **argv )
...
-  app.setThemeName( settings.value( "/Themes", "default" ).toString() );
 
 #ifdef RELATIVE_PREFIX_PATH
+  app.setPrefixPath( app.applicationDirPath() + "/..", true );
   qputenv( "GDAL_DATA", QDir::toNativeSeparators( app.applicationDirPath() + "/../share/gdal" ).toLocal8Bit() );
   const QString projPath( QDir::toNativeSeparators( app.applicationDirPath() + "/../share/proj" ) );
-  app.setPrefixPath( app.applicationDirPath() + "/..", true );
 #else
-  const QString projPath;
-  app.setPrefixPath( CMAKE_INSTALL_PREFIX, true );
```
Comment (nirvn) [inline src/app/main.cpp:216]: @m-kuhn , if I revert this change, I get QField to properly load QGIS providers et cie.

Comment (m-kuhn) [reply src/app/main.cpp:216]: Confirmed, let me think of a better solution here

Comment (m-kuhn) [reply src/app/main.cpp:216]: FindQGIS.cmake is updated, it will now set `QGIS_PREFIX_PATH` and assumes this is at `location_of_qgiscore_library/../..`. To be confirmed if this breaks other builds

Comment (nirvn) [reply src/app/main.cpp:216]: Nice. Will test within the hour.

**Comment by m-kuhn** [comment]
App crashes on start

**Comment by m-kuhn** [comment]
Proj db missing

**Comment by nirvn** [comment]
@m-kuhn , just tried this on my laptop, I can these errors:


```
  Could NOT find QGIS (missing: QGIS_Core_LIBRARY QGIS_Analysis_LIBRARY
  QGIS_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindQGIS.cmake:110 (find_package_handle_standard_args)
  CMakeLists.txt:128 (find_package)
```

I see a QGIS_CORE_LIBRARY variable set, so maybe this is just a capitalization issue?

**Comment by nirvn** [comment]
@m-kuhn , using QGIS_PREFIX_PATH (which is set to match QGIS' CMAKE_INSTALL_PREFIX)

**Comment by m-kuhn** [comment]
Can you try `QGIS_ROOT` instead?

**Comment by nirvn** [comment]
@m-kuhn , that worked, QField now builds. No providers in sight though:
image

**Comment by m-kuhn** [comment]
Can you check the generated qfield.h file in the build dir?

https://github.com/opengisch/QField/pull/3117/files#diff-6197d7a7fceab50dcace262dda32978137b4eb326af033dcb403e6ccbc36de49R64

**Comment by nirvn** [comment]
@m-kuhn , good thinking -- the value is wrong. Here's what is in the qfield.hj:

`#define QGIS_PREFIX_PATH "/home/webmaster"`

The value of QGIS_ROOT is "/home/webmaster/apps" so somehow the qfield.h gets the parent folder of the actual QGIS root/prefix path.

**Comment by nirvn** [comment]
I see XYZ raster tiles rendering in QField now, bravo sir.

**Comment by m-kuhn** [comment]
I think this is ready.
Would be good to have a ios test device verifying sample projects are fixed (i.e. show up on fresh install and can be edited without errors).
@3nids or @lucienicolier can you do that?

**Comment by m-kuhn** [comment]
@nirvn can you check if the latest commit builds fine for you (i.e. if it still finds qtkeychain which tries to find all different spelling patterns across platforms)

**Comment by m-kuhn** [comment]
... and if someone can confirm that the ios sample_projects now show up we are ready to merge this IMHO

**Comment by m-kuhn** [comment]
Sample projects working 💪

**Comment by nirvn** [comment]
@m-kuhn , compiles perfectly fine on my linux workstation, and the android APK works flawlessly on my phone. It seems, we, are, there! :)

**Comment by m-kuhn** [comment]
Thanks for testing. Codewise, are you happy too?

**Review by nirvn — APPROVED** [review]
Yep
